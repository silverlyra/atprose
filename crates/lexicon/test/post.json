{
  "lexicon": 1,
  "id": "dev.atprose.test.post",
  "defs": {
    "main": {
      "type": "record",
      "description": "A post on some service (for testing)",
      "key": "tid",
      "record": {
        "type": "object",
        "required": [
          "id",
          "body"
        ],
        "properties": {
          "id": {
            "type": "string",
            "description": "unique ID"
          },
          "body": {
            "type": "ref",
            "ref": "#body"
          },
          "createdAt": {
            "type": "string",
            "format": "datetime"
          }
        }
      }
    },
    "body": {
      "type": "object",
      "required": [
        "text"
      ],
      "properties": {
        "text": {
          "type": "string",
          "maxLength": 3000,
          "maxGraphemes": 300
        },
        "languages": {
          "type": "array",
          "maxLength": 3,
          "items": {
            "type": "string",
            "format": "language",
            "maxLength": 5
          }
        }
      }
    }
  }
}
