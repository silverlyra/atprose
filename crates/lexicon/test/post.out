Document {
    version: Version(
        1,
    ),
    id: Nsid {
        authority: "dev.atprose.test",
        package: "post",
    },
    revision: None,
    metadata: Metadata {
        description: None,
    },
    defs: {
        "main": Record(
            Record {
                metadata: Metadata {
                    description: Some(
                        "A post on some service (for testing)",
                    ),
                },
                key: Tid,
                def: Object(
                    Object {
                        metadata: Metadata {
                            description: None,
                        },
                        properties: {
                            "id": String(
                                String {
                                    metadata: Metadata {
                                        description: Some(
                                            "unique ID",
                                        ),
                                    },
                                    format: None,
                                    default: None,
                                    value: None,
                                    min_length: None,
                                    max_length: None,
                                    min_graphemes: None,
                                    max_graphemes: None,
                                    known_values: None,
                                    values: None,
                                },
                            ),
                            "body": Ref(
                                Ref {
                                    metadata: Metadata {
                                        description: None,
                                    },
                                    target: RefTarget {
                                        ns: None,
                                        name: Some(
                                            "body",
                                        ),
                                    },
                                },
                            ),
                            "createdAt": String(
                                String {
                                    metadata: Metadata {
                                        description: None,
                                    },
                                    format: Some(
                                        Datetime,
                                    ),
                                    default: None,
                                    value: None,
                                    min_length: None,
                                    max_length: None,
                                    min_graphemes: None,
                                    max_graphemes: None,
                                    known_values: None,
                                    values: None,
                                },
                            ),
                        },
                        required: [
                            "id",
                            "body",
                        ],
                        nullable: [],
                    },
                ),
            },
        ),
        "body": Object(
            Object {
                metadata: Metadata {
                    description: None,
                },
                properties: {
                    "text": String(
                        String {
                            metadata: Metadata {
                                description: None,
                            },
                            format: None,
                            default: None,
                            value: None,
                            min_length: None,
                            max_length: Some(
                                3000,
                            ),
                            min_graphemes: None,
                            max_graphemes: Some(
                                300,
                            ),
                            known_values: None,
                            values: None,
                        },
                    ),
                    "languages": Array(
                        Array {
                            metadata: Metadata {
                                description: None,
                            },
                            items: String(
                                String {
                                    metadata: Metadata {
                                        description: None,
                                    },
                                    format: Some(
                                        Language,
                                    ),
                                    default: None,
                                    value: None,
                                    min_length: None,
                                    max_length: Some(
                                        5,
                                    ),
                                    min_graphemes: None,
                                    max_graphemes: None,
                                    known_values: None,
                                    values: None,
                                },
                            ),
                            min_length: None,
                            max_length: Some(
                                3,
                            ),
                        },
                    ),
                },
                required: [
                    "text",
                ],
                nullable: [],
            },
        ),
    },
}
